scripts/zones/Southern_San_dOria_[S]/npcs/Saphiriance_TK.lua

```diff
@@ -2,20 +2,34 @@
 -- Area: Southern SandOria [S]
 --  NPC: Saphiriance T.K
 -- !pos 113 1 -40 80
+-- Retrace NPC
+-----------------------------------
+require("scripts/globals/campaign")
+require("scripts/globals/teleports")
 -----------------------------------
 local entity = {}
 
 entity.onTrade = function(player, npc, trade)
 end
 
 entity.onTrigger = function(player, npc)
-    player:startEvent(450)
+    local allegiance =  player:getCampaignAllegiance()
+    local rank = getMedalRank(player)
```

Comment:
Just noticed one more thing, could you comment out rank since its unused at this time?

Reply:
Just nuked it for now.